diff --git a/gen_wrap.py b/gen_wrap.py
index 3c853fb..79b7b78 100644
--- a/gen_wrap.py
+++ b/gen_wrap.py
@@ -1466,7 +1466,7 @@ def write_exposer(
             f"       new (t) isl::{wrap_class}(result);"
             "    else"
             f'       isl::handle_isl_error(ctx, "isl_{meth.cls}_read_from_str");'
-            '}, py::arg("s"), py::arg("context").none(true)=py::none());\n')
+            '}, py::arg("s"), py::arg("context").none()=py::none());\n')
 
 # }}}
 
diff --git a/src/wrapper/wrap_isl_part1.cpp b/src/wrapper/wrap_isl_part1.cpp
index 21c1e1f..ab2fe9d 100644
--- a/src/wrapper/wrap_isl_part1.cpp
+++ b/src/wrapper/wrap_isl_part1.cpp
@@ -101,7 +101,7 @@ void islpy_expose_part1(py::module_ &m)
           new (t) isl::val(result);
         else
           isl::handle_isl_error(ctx, "isl_val_from_si");
-      }, py::arg("i"), py::arg("context").none(true)=py::none()
+      }, py::arg("i"), py::arg("context").none()=py::none()
       );
 
   MAKE_WRAP(multi_val, MultiVal);
@@ -151,8 +151,8 @@ void islpy_expose_part1(py::module_ &m)
         else
           isl::handle_isl_error(ctx, "isl_id_alloc");
       }, py::arg("name"),
-      py::arg("user").none(true)=py::none(),
-      py::arg("context").none(true)=py::none()
+      py::arg("user").none()=py::none(),
+      py::arg("context").none()=py::none()
       );
   wrap_id.def("__eq__", islpy::id_eq, py::arg("other"),
       "__eq__(self, other)\n\n"
