diff --git a/config/BuildSystem/config/packages/PnetCDF.py b/config/BuildSystem/config/packages/PnetCDF.py
index e60dad44caf..1cd6a8c4191 100644
--- a/config/BuildSystem/config/packages/PnetCDF.py
+++ b/config/BuildSystem/config/packages/PnetCDF.py
@@ -24,6 +24,28 @@ class Configure(config.package.GNUPackage):
     self.deps  = [self.mpi,self.flibs]
     return
 
+  def applyPatches(self):
+    try:
+      patch_url = 'https://github.com/Parallel-NetCDF/PnetCDF/commit/2b48c9bbc4644cb3ec41731ae938ea0cb5351b73.patch'
+      patch_file = os.path.join(self.packageDir, 'pnetcdf-fix.patch')
+
+      # Download patch
+      if not os.path.exists(patch_file):
+        import urllib.request
+        self.logPrint('Downloading patch from %s' % patch_url)
+        with urllib.request.urlopen(patch_url) as response, open(patch_file, 'wb') as out_file:
+          out_file.write(response.read())
+
+      # Apply patch
+      self.logPrint('Applying patch %s' % patch_file)
+      output,err,ret  = config.package.GNUPackage.executeShellCommand(
+                          'patch -p1 < "%s"' % patch_file,
+                          cwd=self.packageDir, log=self.framework.log)
+      if ret:
+        raise RuntimeError('Error applying patch: %s' % err)
+    except Exception as e:
+      raise RuntimeError('Unable to apply PnetCDF patch: %s' % str(e))
+
   def formGNUConfigureArgs(self):
     # https://github.com/Parallel-NetCDF/PnetCDF/commit/38d210c006cabff70d78204d2db98a22ab87547c
     if hasattr(self.mpi,'ompi_version') and self.mpi.ompi_version >= (4,0,0):
